YAFCmodel/Analysis/Analysis.cs

```diff
@@ -12,7 +12,7 @@ public abstract class Analysis {
         }
 
         public static void ProcessAnalyses(IProgress<(string, string)> progress, Project project, ErrorCollector errors) {
-            foreach (var analysis in analyses) {
+            foreach (Analysis analysis in analyses) {
```

Comment:
I know that you decided to use explicit types instead of var, but in a foreach loop that will perform a cast and might throw InvalidCastException. Which makes it easy to introduce subtle bugs in refactorings.
In my experience its better to rely on the implicit typing in foreach loops and let the compiler do its magic type inference.

Reply:
`private static readonly List<Analysis> analyses = new List<Analysis>();`
I'll need some time to make a solid argument regarding `var` conversion and possible type-cast shenanigans.